Gems/ScriptCanvas/Code/Include/ScriptCanvas/Core/Slot.h

```diff
@@ -149,7 +157,12 @@ namespace ScriptCanvas
 
         bool CanConvertToReference(bool isNewSlot = false) const;
         bool ConvertToReference(bool isNewSlot = false);
-        void SetVariableReference(const VariableId& variableId);
+        enum class IsVariableTypeChange
```

Comment:
minor: why do we use enum here instead of boolean?

Reply:
Code readability. changeGears(false); doesn't tell the reader anything, it forces them to look up the meaning of the bool. changeGears(Method::ManualTransmission); communicates more.